integration/dkg/dkg_test.go

```diff
@@ -0,0 +1,248 @@
+package dkg
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	dkgeng "github.com/onflow/flow-go/engine/consensus/dkg"
+	"github.com/onflow/flow-go/engine/testutil"
+	engmock "github.com/onflow/flow-go/engine/testutil/mock"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/dkg"
+	"github.com/onflow/flow-go/network/stub"
+	"github.com/onflow/flow-go/state/protocol/events/gadgets"
+	protocolmock "github.com/onflow/flow-go/state/protocol/mock"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/storage/badger"
+	"github.com/onflow/flow-go/utils/unittest"
+	"github.com/onflow/flow-go/utils/unittest/mocks"
+)
+
+// node is an in-process node that only contains the engines relevant for DKG,
+// ie. MessagingEngine and ReactorEngine
+type node struct {
+	engmock.GenericNode
+	keyStorage      storage.DKGKeys
+	messagingEngine *dkgeng.MessagingEngine
+	reactorEngine   *dkgeng.ReactorEngine
+}
+
+func (n node) Ready() {
+	<-n.messagingEngine.Ready()
+	<-n.reactorEngine.Ready()
+}
+
+func (n node) Done() {
+	<-n.messagingEngine.Done()
+	<-n.reactorEngine.Done()
+}
+
+// create a set of nodes that share the same hub for networking, the same
+// whiteboard for exchanging DKG broadcast messages, and a mocked state
+// containing the expected next EpochSetup event
+func createNodes(
+	t *testing.T,
+	hub *stub.Hub,
+	chainID flow.ChainID,
+	whiteboard *whiteboard,
+	conIdentities flow.IdentityList,
+	epochSetup flow.EpochSetup,
+	firstBlockID flow.Identifier) ([]*node, flow.IdentityList) {
+
+	// We need to initialise the nodes with a list of identities that contain
+	// all roles, otherwise there would be an error initialising the first epoch
+	colIdentity := unittest.IdentityFixture(unittest.WithRole(flow.RoleCollection))
+	exeIdentity := unittest.IdentityFixture(unittest.WithRole(flow.RoleExecution))
+	verIdentity := unittest.IdentityFixture(unittest.WithRole(flow.RoleVerification))
+	identities := append(conIdentities, colIdentity, exeIdentity, verIdentity)
```

Comment:
We have a util for this
```suggestion
	identities := unittest.CompleteIdentitySet(identities...)
```